sympy/polys/numberfields/numbers.py

```diff
@@ -0,0 +1,96 @@
+"""Algebraic numbers -- forms of expression, and isolation."""
+
+from sympy import AlgebraicNumber, sympify
+from sympy.polys.numberfields.isomorphism import field_isomorphism
+from sympy.polys.numberfields.minpoly import minpoly, primitive_element
+from sympy.polys.polyerrors import IsomorphismFailed
+from sympy.printing.lambdarepr import LambdaPrinter
+from sympy.printing.pycode import PythonCodePrinter, MpmathPrinter
+from sympy.utilities import (
+    lambdify, public
+)
+
+from mpmath import mp
+
+
+@public
+def to_number_field(extension, theta=None, *, gen=None):
+    """Express `extension` in the field generated by `theta`. """
+    if hasattr(extension, '__iter__'):
+        extension = list(extension)
+    else:
+        extension = [extension]
+
+    if len(extension) == 1 and isinstance(extension[0], tuple):
+        return AlgebraicNumber(extension[0])
+
+    minpoly, coeffs = primitive_element(extension, gen, polys=True)
+    root = sum([ coeff*ext for coeff, ext in zip(coeffs, extension) ])
+
+    if theta is None:
+        return AlgebraicNumber((minpoly, root))
+    else:
+        theta = sympify(theta)
+
+        if not theta.is_AlgebraicNumber:
+            theta = AlgebraicNumber(theta, gen=gen)
+
+        coeffs = field_isomorphism(root, theta)
+
+        if coeffs is not None:
+            return AlgebraicNumber(theta, coeffs)
+        else:
+            raise IsomorphismFailed(
+                "%s is not in a subfield of %s" % (root, theta.root))
+
+
+class IntervalPrinter(MpmathPrinter, LambdaPrinter):
+    """Use ``lambda`` printer but print numbers as ``mpi`` intervals. """
```

Comment:
I realise that this PR is just moving this into a package but I really don't understand why this class is defined here.